### PR TITLE
Jo python3 compat

### PR DESCRIPTION
**Pull request by joroscoSF**
## What does this PR do?

apt-transport-s3 is now requires python3
Removed 3rd party dependency

## Why is this change being made?

Taking steps towards a Python3 world

## How does this PR do it?

Rejiggers and refactors away Python2 code for the new fancy Python3 stuff

## How was this tested? How can the reviewer verify your testing?

Docker container for now.  Once this gets merged I'll update the fpm-recipe to build a new package and then test more thoroughly using the promotion process.

## What gif best describes this PR or how it makes you feel?

alt text

### CODE REVIEW

File: s3 [hunk 1]
```diff
@@ -73,10 +69,10 @@ class AWSCredentials(object):
                     raise e
             except socket.timeout:
                 time.sleep(wait_time(i + 1))
-            except urllib2.URLError as e:
+            except urllib.error.URLError as e:
                 if hasattr(e, 'reason'):
-                    raise Exception("URL error reason: %s, probable cause is that\
-     you don't have IAM role on this machine" % e.reason)
+                    raise Exception("URL error reason: %s, probable cause is that you don't have IAM role on this",
```
Comment (paul-krohn) [inline s3]: `... don't have an IAM role ...`

File: s3 [hunk 2]
```diff
@@ -111,35 +114,35 @@ class AWSCredentials(object):
...
 
-        self.region = data.get('Region', 'us-east-1')
+        self.region = self.sanitize(data['auth']['Region'])
         if self.region is None or self.region == '':
             raise Exception("Region required")
         sys.stderr.write("Region: {}\n".format(self.region))
 
-        if data.get("AccessKeyId") is not None:
+        if data['auth']['AccessKeyId'] is not None:
             sys.stderr.write("Using config file for credentials\n")
```
Comment (paul-krohn) [inline s3]: Is it possible to suppress this output?

File: s3 [hunk 3]
```diff
@@ -111,35 +114,35 @@ class AWSCredentials(object):
...
-            data['AccessKeyId'] = os.environ.get("AWS_ACCESS_KEY_ID", None)
-            data['SecretAccessKey'] = os.environ.get("AWS_SECRET_ACCESS_KEY", None)
-            data['Token'] = os.environ.get("AWS_SESSION_TOKEN", None)
-            if data.get("AccessKeyId") is not None:
+        if data['auth']['AccessKeyId'] is None:
+            data['auth']['AccessKeyId'] = os.environ.get("AWS_ACCESS_KEY_ID", None)
+            data['auth']['SecretAccessKey'] = os.environ.get("AWS_SECRET_ACCESS_KEY", None)
+            data['auth']['Token'] = os.environ.get("AWS_SESSION_TOKEN", None)
+            if data['auth']['AccessKeyId'] is not None:
                 sys.stderr.write("Using environment variables for credentials\n")
```
Comment (paul-krohn) [inline s3]: this too ...

File: s3 (hunk 1 above)
Comment (joroscoSF) [reply s3]: This doesn't and probably never will apply to us, it's wrapped in an  If statement we never drop into.

**Pull request merged**